src/textevent.cpp

```diff
@@ -107,7 +110,29 @@ TextEvent::TextEvent(const QString &accountId,
                    InformationType informationType,
                    const TextEventAttachments &attachments,
                    const Participants &participants)
-    : Event(*new TextEventPrivate(accountId, threadId, eventId, sender, timestamp, newEvent,
+    : Event(*new TextEventPrivate(accountId, threadId, eventId, sender, timestamp, sentTime, newEvent,
+                                  message, messageType, messageStatus, readTimestamp, subject, informationType,
+                                  attachments, participants))
+{
+    qDBusRegisterMetaType<QList<QVariantMap> >();
+    qRegisterMetaType<QList<QVariantMap> >();
+}
+
+TextEvent::TextEvent(const QString &accountId,
+                   const QString &threadId,
+                   const QString &eventId,
+                   const QString &sender,
+                   const QDateTime &timestamp,
+                   bool newEvent,
+                   const QString &message,
+                   MessageType messageType,
+                   MessageStatus messageStatus,
+                   const QDateTime &readTimestamp,
+                   const QString &subject,
+                   InformationType informationType,
+                   const TextEventAttachments &attachments,
+                   const Participants &participants)
+    : Event(*new TextEventPrivate(accountId, threadId, eventId, sender, timestamp, timestamp, newEvent,
```

Comment:
What will happen if this contructor without a sentTime is called, then later the sentTime is requested on the TextEvent object?

Reply:
The sent date will have the value of the received date